### DISCUSSION TITLE
SEO description not working properly

### DISCUSSION THREAD

**Discussion by BobTheExplorer** [mmistakes/minimal-mistakes#3151] [Q&A] [answered]
On search engine results my yaml description does not show rather a jumble of words from the main page of the site, I tried putting description and excerpt in but it does not change anything whatsoever

**Comment by mmistakes** [comment, marked as answer]
Do you have an example? It’s quite common for Google to rewrite your page titles and descriptions to make them more relevant to a search query. It doesn’t always do a good job of that though…

In fact there has been a lot of noise about this recently in the SEO community as Google changed how they deal with title tags.

it’s also worth noting that just because you populate a page’s meta description, doesn’t mean Google or other search engines will show that text in results. There’s a good chance you’ve done everything correct, Google is just ignoring it… which isn’t uncommon.

**Reply by BobTheExplorer** [reply]
Well for my index.html page I put both a meta and regular description in their respective places. Though it get's overlooked as when I google my site in any engine it displays text from the page, ie search or recent posts.

**Reply by mmistakes** [reply]
By example I meant a URL to the web page and the source code to it in some sort of public repo so it can be inspected.

**Reply by mmistakes** [reply]
Also you might want to have a read through this article by Google on how they use titles.
https://developers.google.com/search/blog/2021/08/update-to-generating-page-titles

They don't always use the content a page provides in the `meta` "tags", and recently Google started pulling titles from the `h1` element instead of the `title` in a lot of cases.

Moz also did a write up and analysis on this and found examples of where Google was producing worse title's in SERPs for certain pages.

https://moz.com/blog/ways-google-rewrites-title-tags

TL/DR... there might not be anything you can do about it.

**Reply by om-hb** [reply]
I also noticed that Google seems to like to use the headings from the ToC as page description, as can be seen here:
![grafik](https://user-images.githubusercontent.com/49337083/140952616-6220db3a-5727-4c5c-a817-d2d64a59c7cd.png)

I guess this is also a result of the changes you mentioned?

**Reply by mmistakes** [reply]
Yup. Theme is working fine and as expected. Google sometimes thinks it knows better than the H1 or `title` you define and they calculate the title to something else in their search results.

Welcome to the fun and exciting world of SEO!?!

**Comment by Rijaspb** [comment]
https://www.rijasexports.com/
